tests/e2e/components_test.go

```diff
@@ -0,0 +1,280 @@
+		client.MatchingLabels{
+			labels.PlatformPartOf: strings.ToLower(c.GVK.Kind),
+		},
+	).Eventually().ShouldNot(
+		BeEmpty(),
+	)
+
+	for _, d := range deployments {
+		t.Run("deployment_"+d.GetName(), func(t *testing.T) {
+			replicas, err := jq.ExtractValue[int](d, `.spec.replicas`)
+			g.Expect(err).ShouldNot(HaveOccurred())
+
+			expectedReplica := replicas + 1
+			if replicas > 1 {
+				expectedReplica = 1
+			}
+
+			g.Update(
+				gvk.Deployment,
+				client.ObjectKeyFromObject(&d),
+				testf.Transform(`.spec.replicas = %d`, expectedReplica),
+			).Eventually().WithTimeout(30 * time.Second).WithPolling(1 * time.Second).Should(
+				jq.Match(`.spec.replicas == %d`, expectedReplica),
+			)
+
+			g.Get(
+				gvk.Deployment,
+				client.ObjectKeyFromObject(&d),
+			).Eventually().Should(
+				jq.Match(`.spec.replicas == %d`, expectedReplica),
+			)
+
+			g.Get(
+				gvk.Deployment,
+				client.ObjectKeyFromObject(&d),
+			).Consistently().WithTimeout(30 * time.Second).WithPolling(1 * time.Second).Should(
+				jq.Match(`.spec.replicas == %d`, expectedReplica),
+			)
+		})
+	}
+}
+
+func (c *ComponentTestCtx) ValidateComponentDisabled(t *testing.T) {
+	g := c.NewWithT(t)
+
+	g.List(c.GVK).Eventually().ShouldNot(
+		BeEmpty(),
+	)
+
+	g.Update(
+		gvk.DataScienceCluster,
+		c.DSCName,
+		testf.Transform(`.spec.components.%s.managementState = "%s"`, strings.ToLower(c.GVK.Kind), operatorv1.Removed),
+	).Eventually().Should(
+		jq.Match(`.spec.components.%s.managementState == "%s"`, strings.ToLower(c.GVK.Kind), operatorv1.Removed),
+	)
+
+	g.List(c.GVK).Eventually().Should(
+		BeEmpty(),
+	)
+
+	g.List(
+		gvk.Deployment,
+		client.InNamespace(c.ApplicationNamespace),
+		client.MatchingLabels{
+			labels.PlatformPartOf: strings.ToLower(c.GVK.Kind),
+		},
+	).Eventually().Should(
+		BeEmpty(),
+	)
+
+	g.List(gvk.DataScienceCluster).Eventually().Should(And(
+		HaveLen(1),
+		HaveEach(And(
+			jq.Match(`.spec.components.%s.managementState == "%s"`, strings.ToLower(c.GVK.Kind), operatorv1.Removed),
+			jq.Match(`.status.conditions[] | select(.type == "%sReady") | .status == "%s"`, c.GVK.Kind, metav1.ConditionFalse),
+		)),
+	))
+}
+
+func (c *ComponentTestCtx) ValidateCRDReinstated(t *testing.T, name string) {
+	t.Helper()
+
+	g := c.NewWithT(t)
+	crdSel := client.MatchingFields{"metadata.name": name}
+
+	g.Update(
+		gvk.DataScienceCluster,
+		c.DSCName,
+		testf.Transform(`.spec.components.%s.managementState = "%s"`, strings.ToLower(c.GVK.Kind), operatorv1.Removed),
+	).Eventually().Should(
+		jq.Match(`.spec.components.%s.managementState == "%s"`, strings.ToLower(c.GVK.Kind), operatorv1.Removed),
+	)
+
+	g.List(c.GVK).Eventually().Should(
+		BeEmpty(),
+	)
+	g.List(gvk.CustomResourceDefinition, crdSel).Eventually().Should(
+		HaveLen(1),
+	)
+
+	g.Delete(
+		gvk.CustomResourceDefinition,
+		types.NamespacedName{Name: name},
+		client.PropagationPolicy(metav1.DeletePropagationForeground),
+	).Eventually().ShouldNot(
```

Comment:
This check of implicit error after delete makes the DSL a bit confusing. For List works fine, for Get so/so, but for Delete like opposite. But I do not have solution I would really like in my mind at the moment.

Reply:
seems so, updated